astropy/units/format/base.py

```diff
@@ -24,15 +26,96 @@ def __init_subclass__(cls, **kwargs):
         super().__init_subclass__(**kwargs)
 
     @classmethod
-    def parse(cls, s):
+    def _get_unit_name(cls, unit):
+        return unit.get_format_name(getattr(cls, "name", None))
+
+    @classmethod
+    def format_exponential_notation(cls, val, format_spec="g"):
         """
-        Convert a string to a unit object.
+        Formats a value in exponential notation.
+
+        Parameters
+        ----------
+        val : number
+            The value to be formatted
+
+        format_spec : str, optional
+            Format used to split up mantissa and exponent
+
+        Returns
+        -------
+        string : str
+            The value in exponential notation in a this class's format.
         """
-        raise NotImplementedError(f"Can not parse with {cls.__name__} format")
+        return format(val, format_spec)
+
+    @classmethod
+    def _format_superscript(cls, number):
+        return f"({number})" if "/" in number or "." in number else number
 
     @classmethod
-    def to_string(cls, u):
+    def _format_unit_power(cls, unit, power=1):
+        """Format the unit for this format class raised to the given power.
+
+        This is overridden in Latex where the name of the unit can depend on the power
+        (e.g., for degrees).
         """
-        Convert a unit object to a string.
+        name = cls._get_unit_name(unit)
+        if power != 1:
+            name += cls._format_superscript(utils.format_power(power))
+        return name
+
+    @classmethod
+    def _format_unit_list(cls, units):
```

Comment:
I think a tuple will always work here. Perhaps this should be changed since tuples are faster and more memory efficient.

Reply:
Agreed. Did you think we should change the name? (I think better to leave it the same)

Reply:
It's private, so we're free to change it. And it's more explicit and will help when we add typing. I would be inclined to change the name.

Reply:
`utils.grouped_by_powers` does return lists for `nominator` and `denominator` (constructed by appending, so logical). So, `_format_unit_tuple` is incorrect. 

Actually, looking at the override in`generic` - that does a `.sort`, so it does assume a list.  Frankly, `generic` with `inline=True` looks awful, but that is for the next PR... For now, let me just revert those tuples to lists (and add a test...)

Reply:
SGTM. I suppose allowing lists, but using tuples wherever possible is good enough.